### PR TITLE
when suggesting RUST_BACKTRACE=1, add a special note for Miri's env var isolation

### LINKED ISSUE #2855: By default, RUST_BACKTRACE=1 silently has no effect on Miri, even though it is suggested in panic messages
cc https://github.com/rust-lang/rust/pull/98112 
cc @saethlin 

Code from the pr
````rust
fn main() {
    let mut x = [0u32; 2];
    let ptr: *mut u8 = x.as_mut_ptr().cast::<u8>();
    unsafe {
        *(ptr.add(1).cast::<u32>()) = 42;
    }
}
````

run with `RUST_BACKTRACE=1 ~/.cargo/bin/cargo miri run` seems to not print any backtrace disregarding what the panic output suggests:

````
Preparing a sysroot for Miri (target: x86_64-unknown-linux-gnu)... done
WARNING: Ignoring `RUSTC_WRAPPER` environment variable, Miri does not support wrapping.
    Finished dev [unoptimized + debuginfo] target(s) in 0.00s
     Running `/home/matthias/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/cargo-miri runner target/miri/x86_64-unknown-linux-gnu/debug/f`
thread 'main' panicked at 'misaligned pointer dereference: address must be a multiple of 0x4 but is 0x278b1', src/main.rs:5:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

````

### PR DESCRIPTION
**Pull request by RalfJung**
Fixes https://github.com/rust-lang/miri/issues/2855

### CODE REVIEW

**Review by Noratrieb — APPROVED** [review]

**Comment by workingjubilee** [comment]
Huh.

@RalfJung Would you rather we always print a backtrace when we panic while running in Miri? That was one of the solutions discussed in https://github.com/rust-lang/miri/issues/2855

Note this is different from "setting `RUST_BACKTRACE=1`", it won't affect user library behavior.

**Comment by workingjubilee** [comment]
Well, approximately-equal-to one of the solutions.

**Comment by RalfJung** [comment]
> @RalfJung Would you rather we always print a backtrace when we panic while running in Miri? That was one of the solutions discussed in https://github.com/rust-lang/miri/issues/2855

I don't think that would be a good default for Miri, just like it is not a good default for regular execution.

**Comment by workingjubilee** [comment]
:thinking: It would seem to me you usually would want a backtrace? But okay!

**Comment by saethlin** [comment]
> just like it is not a good default for regular execution.

Funny you should say that. Every codebase at my employer has this at the top of `main`:
```rust
    if std::env::var("RUST_BACKTRACE").is_err() {
        std::env::set_var("RUST_BACKTRACE", "1");
    }    
```
And I know from a discussion on Mastodon a few days ago that we are not the only ones doing this.

**Comment by workingjubilee** [comment]
Yes. I think we _would_ want to default to it on, except we should get better at printing shorter backtraces, so that `RUST_BACKTRACE=1` strikes the right balance of output and informative.

**Comment by RalfJung** [comment]
Okay, let me rephrase: I don't think Miri's default should be different from that of regular Rust.

**Comment by workingjubilee** [comment]
Ah, I see! I only figured that since Miri strongly functions as an error-reporting tool, that you might consider deviating on this point acceptable. But I do understand preferring to avoid differences in execution, considering it mostly is good at error-reporting by being good at simulation.

**Comment by RalfJung** [comment]
Almost every time we made Miri divert from regular execution, we eventually decided that was a bad idea. Maybe this is one of the few exceptions, but it doesn't seem obvious to me.

**Comment by saethlin** [comment]
Yeah; on balance I think the extra note is the best approach.
